test/automation/src/positron/positronEditors.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from '@playwright/test';
+import { Code } from '../code';
+
+
+export class PositronEditors {
+
+	activeEditor = this.code.driver.page.locator('div.tab.tab-actions-right.active.selected');
+	editorIcon = this.code.driver.page.locator('.monaco-icon-label.file-icon');
+	editorPart = this.code.driver.page.locator('.split-view-view .part.editor');
+
+	constructor(private code: Code) { }
+
+	async waitForActiveTab(fileName: string, isDirty: boolean = false): Promise<void> {
+		await expect(this.code.driver.page.locator(`.tabs-container div.tab.active${isDirty ? '.dirty' : ''}[aria-selected="true"][data-resource-name$="${fileName}"]`)).toBeVisible();
+	}
+
+	async newUntitledFile(): Promise<void> {
+		if (process.platform === 'darwin') {
+			await this.code.driver.page.keyboard.press('Meta+N');
+		} else {
+			await this.code.dispatchKeybinding('Control+N');
+		}
+
+		await this.waitForEditorFocus('Untitled-1');
+	}
+
+	async waitForEditorFocus(fileName: string): Promise<void> {
+		await this.waitForActiveTab(fileName, undefined);
+		await this.waitForActiveEditor(fileName);
+	}
+
+	async waitForActiveEditor(fileName: string): Promise<any> {
+		const selector = `.editor-instance .monaco-editor[data-uri$="${fileName}"] textarea`;
+		await expect(this.code.driver.page.locator(selector)).toBeFocused();
+	}
+
+	async selectTab(fileName: string): Promise<void> {
+
+		// Selecting a tab and making an editor have keyboard focus
+		// is critical to almost every test. As such, we try our
+		// best to retry this task in case some other component steals
+		// focus away from the editor while we attempt to get focus
+
+		let error: unknown | undefined = undefined;
+		let retries = 0;
+		while (retries < 10) {
```

Comment:
Could this be the "expect" toPass() instead?

Reply:
yeah, i do like that the basic loops can capture the error and surface it, but this could definitely be a toPass().  i could go either way.  it is a little annoying that toPass blocks always just fail on a timeout.

Reply:
i will go ahead and convert.... toPass is more concsie